File: frontend/lib/types/monthly-mvp.ts
/** Tipos que mapean las respuestas de /api/admin/months/* (UC-17) */

export interface MonthlyCandidate {
  player_id: number;
  display_name: string;
  area: string;
  mvp_cycles: string[];
  mvp_cycle_ids: number[];
  sp_total_month: number;
  cp_total_month: number;
  subtasks_done_month: number;
  avg_m_calidad: number | null;
  metric_highlight: string;
}

export interface MonthCycleInfo {
  cycle_id: number;
  name: string;
  status: string;
  mvp_player_id: number | null;
}

export interface MonthKPIs {
  cp_done: number;
  sp_generated: number;
  subtasks_done: number;
  cycles_total: number;
  cycles_closed_or_archived: number;
}

export interface MonthCloseSummary {
  year: number;
  month: number;
  period_label: string;
  cycles: MonthCycleInfo[];
  kpis: MonthKPIs;
  candidates: MonthlyCandidate[];
  already_closed: boolean;
  can_close: boolean;
  blocking_errors: string[];
  warnings: string[];
}

export interface MonthCloseRequest {
  mvp_player_id: number;
  reason: string;
}

export interface MonthlyMvpEditRequest {
  new_player_id: number;
  reason: string;
}

export interface MonthCloseResponse {
  period_label: string;
  player_id: number;
  display_name: string;
  sp_awarded: number;
  ach04_unlocked: boolean;
  message: string;
}

export interface MonthlyMvpHistoryItem {
  id: number;
  period_label: string;
  year: number;
  month: number;
  player_id: number;
  display_name: string;
  area: string | null;
  reason: string | null;
  sp_reward: number;
  assigned_at: string;
  assigned_by_name: string;
  source_cycle_ids: number[];
}

File: backend/src/forge/schemas/monthly_mvp.py
"""Schemas Pydantic para UC-17: cierre mensual y MVP del Mes."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class MonthlyCandidate(BaseModel):
    player_id: int
    display_name: str
    area: str
    mvp_cycles: list[str]
    mvp_cycle_ids: list[int]
    sp_total_month: float
    cp_total_month: int
    subtasks_done_month: int
    avg_m_calidad: float | None
    metric_highlight: str


class MonthCycleInfo(BaseModel):
    cycle_id: int
    name: str
    status: str
    mvp_player_id: int | None


class MonthKPIs(BaseModel):
    cp_done: int
    sp_generated: float
    subtasks_done: int
    cycles_total: int
    cycles_closed_or_archived: int


class MonthCloseSummary(BaseModel):
    year: int
    month: int
    period_label: str
    cycles: list[MonthCycleInfo]
    kpis: MonthKPIs
    candidates: list[MonthlyCandidate]
    already_closed: bool
    can_close: bool
    blocking_errors: list[str]
    warnings: list[str]


class MonthCloseRequest(BaseModel):
    mvp_player_id: int
    reason: str = Field(..., min_length=30, description="Justificación mínimo 30 caracteres")


class MonthlyMvpEditRequest(BaseModel):
    new_player_id: int
    reason: str = Field(..., min_length=30, description="Justificación mínimo 30 caracteres")


class MonthlyMvpHistoryItem(BaseModel):
    id: int
    period_label: str
    year: int
    month: int
    player_id: int
    display_name: str
    area: str | None
    reason: str | None
    sp_reward: int
    assigned_at: datetime
    assigned_by_name: str
    source_cycle_ids: list[int]


class MonthCloseResponse(BaseModel):
    period_label: str
    player_id: int
    display_name: str
    sp_awarded: float
    ach04_unlocked: bool
    message: str
